client/src/components/Dice/DiceContainer.tsx

```diff
@@ -20,13 +21,15 @@ const DiceContainer: React.FC<{ diceState: Record<string, number>, socket: Socke
 	const handleDiceRoll = (): void =>  {
 		socket.emit("roll_dice", {lobbyId});	
 	}
-
+  
+  const isActivePlayer = gameState.activePlayer === userId;
+  
   return (
     <div className="dice-container">
       {diceEntries.map(([colourKey, value]) => (
         <Die key={colourKey} colour={colourMap[colourKey]} colourKey={colourKey} value={value}/>
       ))}
-      <button onClick={handleDiceRoll}>Roll Dice</button>
+      <button onClick={handleDiceRoll} disabled={!isActivePlayer}>Roll Dice</button>
```

Comment:
nah it all works. The turn logic will update the active player in socket and send as part of the serialized gamedata on the frontend. This will then update the button and only enable to button for the active player

Reply:
I meant that the active player can make multiple rolls. We need to implement only allowing a single roll as a separate feature.